dali/pipeline/operator/common.h

```diff
@@ -38,5 +43,130 @@ inline void GetSingleOrRepeatedArg(const OpSpec &spec, vector<T> &result,
       to_string(result.size()) + " given.");
 }
 
+template <typename T>
+void GetPerSampleArgument(std::vector<T> &output,
+                          const std::string &argument_name,
+                          const OpSpec &spec,
+                          const ArgumentWorkspace &ws,
+                          int batch_size = -1 /* -1 = get from "batch_size" arg */) {
```

Comment:
Or I'd rather go ahead and remove the possibility to get batch size from spec. I think we'd like to get rid of fixed batch size at spec level, so I'd say let us not place more obstacles to achieve that later.

Reply:
We still need batch_size arg it in Readers - and VideoReaderResize is one of them.